koap/js_tests/decodeTcp.test.js

```diff
@@ -0,0 +1,39 @@
+let koapModule = require('../build/distributions/koap')
+const koap = koapModule.com.juul.koap
+
+let kotlinModule = require('../build/publications/npm/js/node_modules/kotlin')
+const kotlin = kotlinModule
+
+test('TCP empty payload :: correctly decoded', () => {
+  
+    let encodedMessage = new Uint8Array([0xA1, 0x02, 0x42, 0xB4, 0x69, 0x6E, 0x66, 0x6F, 0x44, 0x62, 0x61, 0x74, 0x74])
+
+    let decodedMessage = koap.decodeTcp(encodedMessage)
+
+    console.log(decodedMessage)
+
+    // Better to not convert? Introduces extra point of failure this way
+    let decodedPayload = byteArrayToHexString(decodedMessage.payload)
```

Comment:
I think I might stop doing this. While byteArrayToHexString has been useful for me while looking at the results, as I commented, it's adding a potential point of failure. I.e. if that function gets a bug, tests will start failing incorrectly.

Reply:
Does JS have assertions that work with `UInt8Array`? If so, then you're right, we should probably stick with it being `UInt8Array` (a.k.a. `ByteArray`) rather than converting it to a String.

Reply:
Yeah, pretty sure jest can check equivalence between two arrays. Gonna change this out.

Reply:
Jest has a deep equality check for this purpose e.g.
```js
let expected = new Uint8Array([...])
expect(decodedMessage.payload).toStrictEqual(expected)
```